### PR TITLE
gajim 1.1.3 (new formula)

### PR DESCRIPTION
**Pull request by amyspark**
- [X] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

This is a replacement for #35919. It installs Gajim with its plugin installer, OMEMO and the image autopreviewer.

I've made sure that `pyobjc` compiles correctly in Sierra. (I've also left the whole posse of `pyobjc` dependencies commented in case you want me to add them. Only `pyobjc`, `pyobjc-core` and the Cocoa framework are needed.)

### CODE REVIEW

**Comment by wurstsalat3000** [comment]
For reference: https://dev.gajim.org/gajim/gajim/issues/9246

File: Formula/gajim.rb
Comment (zbeekman) [inline Formula/gajim.rb]: Can we get a better test here please?

File: Formula/gajim.rb
Comment (zbeekman) [inline Formula/gajim.rb]: Can you please clean up and remove commented out resources?

**Review by zbeekman — CHANGES_REQUESTED** [review]
Some cleanup would be good, I think. Also please squash. Thanks!

File: Formula/gajim.rb
Comment (amyspark) [reply Formula/gajim.rb]: I'm afraid I don't know what else to test, the other options require graphical input and/or an already running instance.

File: Formula/gajim.rb
Comment (amyspark) [inline Formula/gajim.rb]: I didn't include the rest of the hack that appear on other formulae.

**Comment by zbeekman** [comment]
I'm not 100% convinced that this is appropriate for core, as it is effectively just a GUI app, no? But I'll defer to other @Homebrew/maintainers

**Comment by MikeMcQuaid** [comment]
> I'm not 100% convinced that this is appropriate for core, as it is effectively just a GUI app, no? But I'll defer to other @Homebrew/maintainers

This is loosely covered by https://docs.brew.sh/Acceptable-Formulae#stuff-that-builds-a-gui-by-default-but-doesnt-have-to

For Windows they have an actual installer which also seems preferable here (although one doesn't currently exist): https://gajim.org/downloads.php?lang=en#windows

**Comment by MikeMcQuaid** [comment]
Personal preference: I would also like to see us be tentative on new GUI formulae because Homebrew isn't a great delivery mechanism for them. As there's no `.app` on upstream binary though it does seem more useful to have something like this in Homebrew/homebrew-core than to refuse to provide any official way in the Homebrew ecosystem of installing it.

**Comment by zbeekman** [comment]
> I would also like to see us be tentative on new GUI formulae because Homebrew isn't a great delivery mechanism for them.

agreed

> As there's no `.app` on upstream binary though it does seem more useful to have something like this in Homebrew/homebrew-core than to refuse to provide any official way in the Homebrew ecosystem of installing it.

Yeah, in that case let's ship this and try to make a request upstream to provide a macOS binary build?

**Review by zbeekman — APPROVED** [review]

**Comment by slokhorst** [comment]
I'm the one who started the original PR. I wasn't sure either if a Brew formula was appropriate so I asked on [Discourse](https://discourse.brew.sh/t/python-gui-app-formula-core-or-cask/3863) but there wasn't a clear answer there too.

I agree that an .app might be more appropriate, but since there are no experience Mac developers on the Gajim team, this was an easier option for us. If you have any advice on how to turn this into a proper .app, advice is welcome!

**Comment by zbeekman** [comment]
Guys, I've thought about this some more. I really think you should try to further investigate creating and shipping an app. I posted this in the Discourse discussion, which includes some links that I think might get you going. There's a project called `pyinstaller` which is even installable through homebrew and a few good write ups online for how to use it. Because of this, unless there's a CLI version, or a library only version I think I'm actually 👎 on merging this.

Here is what I said on discourse:

---

@slokhorst Is there a CLI only version of the APP that can be built? Or a library? Or is it 100% a gui app written in python?

If it's the latter I really don't think it's appropriate for core.

Check out this post: https://hackernoon.com/the-one-stop-guide-to-easy-cross-platform-python-freezing-part-1-c53e66556a0a

I'd look into pyinstaller which seems capable of creating an app bundle. Then you can make the app bundle and distribute that as a binary.

https://github.com/pyinstaller/pyinstaller

You can even get it through homebrew: `brew install pyinstaller`

**Review by zbeekman — CHANGES_REQUESTED** [review]
See my comments

**Comment by MikeMcQuaid** [comment]
> I'd look into pyinstaller which seems capable of creating an app bundle. Then you can make the app bundle and distribute that as a binary.

Does that handle all necessary native dependencies too?

**Comment by zbeekman** [comment]
> Does that handle all necessary native dependencies too?

I think it does, from my brief survey of it. I'm closing this. If anyone disagrees please let me know and we can re-open.

**Comment by Midar** [comment]
I don't really understand the rationale of why this should not be in homebrew: E.g. easy-tag is a Gtk3 application, and it's also not an .app. How does this differ from Gajim? (I somewhat fear now that the result of this question might be that easy-tag gets removed *sigh*)

**Pull request closed without merging**
